### ISSUE TITLE
h_offset and v_offset for PathFollow3D dont produce intended effect with rotation turned off

### ISSUE THREAD

**Issue by angad-k** [godotengine/godot#46312]
**Godot version:**
`3bb628d8fe26abad258a222a5e22b644e2b73a01`

**OS/device including version:**
`Windows 10 20H2`

**Issue description:**
The h_offset and v_offset for PathFollow3D node are supposed to add a displacement along and perpendicular to the path respectively. However, if we turn off the rotation, it just displaces it along the x and y axes. 

**Steps to reproduce:**
Add a Path3D node and PathFollow3D node as it's child. Define a Path for Path3D and try changing the h_offset and v_offset when rotation is turned off. 

**Minimal reproduction project:**
[hv_offset.zip](https://github.com/godotengine/godot/files/6021521/hv_offset.zip)

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot/issues/3713.

**Comment by precup** [comment]
I don't think it makes sense to have an `h_offset` and `v_offset` for a `PathFollow3D`. Those properties make sense on a `PathFollow2D`, but in 3D you'd need three axes of displacement, and the `h` and `v` names no longer apply. The documentation states that `v_offset` is "The node's offset perpendicular to the curve", but that's not really enough information in 3D since there are infinite options it could be returning.

**Comment by KoBeWi** [comment]
Still valid in Godot 4.5 beta6.

Seems like the culprit is this line:
https://github.com/godotengine/godot/blob/3defc856b0f1c727b4346e60615b2b2451916837/scene/3d/path_3d.cpp#L308
When rotation is disabled, local translation will always have the same orientation.

>  Those properties make sense on a PathFollow2D, but in 3D you'd need three axes of displacement, and the h and v names no longer apply.

Not really. When the current path segment is oriented forward, `h_offset` is oriented facing right and `v_offset` is oriented facing up. You can observe it in the MRP:

https://github.com/user-attachments/assets/cf019eb8-12cd-41aa-9180-074feb1897b4

The third "displacement" is `progress` itself.

**Comment by precup** [comment]
> Not really. When the current path segment is oriented forward, `h_offset` is oriented facing right and `v_offset` is oriented facing up. You can observe it in the MRP:
> godot.windows.editor.dev.x86_64_vHzDfS2aFm.mp4
> 
> The third "displacement" is `progress` itself.

If that's the intention, then this needs a docfix, as well. The current definitions imply that `h_offset` projects along the tangent of the curve because they're just copied from `PathFollow2D`, and in `PathFollow2D` `h_offset` *does* project along the tangent of the curve.
